File: 프로그래머스/1/142086. 가장 가까운 같은 글자/가장 가까운 같은 글자.js
function solution(s) {
    let result = [];
    let temp = [];
    for (let i = 0; i < s.length; i++) {
        if (!temp.includes(s[i])) {
            temp.push(s[i]);
            result.push(-1);
        } else {
            let min = 9999;
            for (let j = i - 1; j >= 0; j--) {
                if (s[i] === s[j]) {
                    if ((i - j) < min) {
                        min = i - j;
                    }
                    temp.push(s[i]);
                    result.push(min);
                    break;
                }
            }
        }
    }
    return result;
}

File: 프로그래머스/1/142086. 가장 가까운 같은 글자/가장 가까운 같은 글자.py
def solution(s):
    result = [];
    temp = [];
    
    for i in range(len(s)):
        if s[i] not in temp:
            temp.append(s[i])
            result.append(-1)
        else:
            minVal = 9999
            for j in range(i - 1, -1, -1):
                if s[i] == s[j]:
                    if minVal > (i - j):
                        minVal = i - j
                    result.append(minVal)
                    temp.append(s[i])
                    break
    return result
                    